client-react/src/pages/app/app-settings/ErrorPages/ErrorPageGrid.tsx

```diff
@@ -2,71 +2,122 @@ import { DetailsListLayoutMode, IColumn, PanelType, SelectionMode, TooltipHost }
   defaultCellStyle,
 } from '../../../../components/DisplayTableWithEmptyMessage/DisplayTableWithEmptyMessage';
 import { PermissionsContext } from '../Contexts';
 import CustomPanel from '../../../../components/CustomPanel/CustomPanel';
 import ErrorPageGridAddEdit from './ErrorPageGridAddEdit';
+import { boldCellStyle } from './ErrorPageGrid.styles';
 
 const ErrorPageGrid: React.FC<FormikProps<AppSettingsFormValues>> = props => {
   const { app_write, editable, saving } = useContext(PermissionsContext);
   const disableAllControls = !app_write || !editable || saving;
   const { t } = useTranslation();
   const [showPanel, setShowPanel] = useState(false);
-  const [currentErrorCode, setCurrentErrorCode] = useState<IColumnItem | null>(null);
+  const [labelAddEditPane, setLabelAddEditPane] = useState<string | undefined>(undefined);
+  const [currentErrorPage, setCurrentErrorPage] = useState<FormErrorPage | null>(null);
+  const [currentErrorPageIndex, setCurrentErrorPageIndex] = useState<number | null>(null);
+  const { values } = props;
+
+  const errorPages = values.errorPages;
 
   const onCancelPanel = React.useCallback((): void => {
     setShowPanel(false);
   }, [setShowPanel, showPanel]);
 
-  const removeItem = React.useCallback((index: number) => {}, []);
+  const removeItem = React.useCallback(
+    (index: number) => {
+      const errorPages: FormErrorPage[] = [...values.errorPages];
+      errorPages.splice(index, 1);
+      props.setValues({
+        ...values,
+        errorPages,
+      });
+    },
+    [values.errorPages]
+  );
 
-  const onShowPanel = React.useCallback((item: IColumnItem, index: number): void => {
-    setShowPanel(true);
-    setCurrentErrorCode(item);
-  }, []);
+  const addEditItem = React.useCallback(
+    (item: FormErrorPage, file: string, index: number) => {
+      const errorPages = [...values.errorPages];
+      item.content = file;
+      if (labelAddEditPane == t('editErrorPage')) {
+        errorPages[index] = item;
+      } else {
+        errorPages.push(item);
+      }
+      props.setValues({
+        ...values,
+        errorPages,
+      });
+      setShowPanel(false);
+    },
+    [values.errorPages, currentErrorPage, currentErrorPageIndex]
+  );
+
+  const onShowPanel = React.useCallback(
+    (item: FormErrorPage, index: number): void => {
+      if (item.status === t('errorPage_columnStatus_configured')) setLabelAddEditPane(t('editErrorPage'));
+      else setLabelAddEditPane(t('addErrorPage'));
+      setShowPanel(true);
+      setCurrentErrorPage(item);
+      setCurrentErrorPageIndex(index);
+    },
+    [values, labelAddEditPane, setLabelAddEditPane, currentErrorPage, currentErrorPageIndex]
+  );
+
+  const getConfigurationStatus = React.useCallback(
+    (errorCode: string) => {
+      if (errorPages.some(i => i.errorCode.includes(errorCode))) return t('errorPage_columnStatus_configured');
+      else return t('errorPage_columnStatus_notConfigured');
```

Comment:
If line 25 removed above:

```suggestion
      return errorPages.some(i => i.errorCode.includes(errorCode))
        ? t('errorPage_columnStatus_configured')
        : t('errorPage_columnStatus_notConfigured');
```